src/com/hotmail/kalebmarc/textfighter/casino/BlackjackGame.java

```diff
@@ -0,0 +1,282 @@
+            if (bet > Coins.get()) {
+                Ui.cls();
+                bet = 0;
+                Ui.println("You do not have enough coins. Please enter a smaller amount. (Or enter 0 to go back)");
+
+            }
+        } while (bet < 10 || bet > 250);
+
+        Coins.set(-bet, true);
+
+        resetCards();
+
+        int playerScore = playerTurn();
+
+        int casinoScore = CasinoTurn();
+
+        // Determine winner
+        Ui.cls();
+        if (playerScore > casinoScore) {
+            Ui.println("You won this round! Congratulations!");
+            return bet * 2;
+        } else if (casinoScore > playerScore) {
+            Ui.println("Unfortunately you lost this time. Good luck next time!");
+            return 0;
+        } else {
+            Ui.println("This round is a match. You get your bet back.");
+            return bet;
+        }
+    }
+
+    /**
+     * Lets the player play his round of blackjack
+     * @return int Score achieved by player. Scores above 21 will result in 0
+     */
+    private int playerTurn() {
+         List<Card> drawnCards = new LinkedList<>();
+         boolean exit = false;
+
+         // Ask the player to draw more cards
+         do{
+             // Draw cards
+             Card drawn = drawCard();
+             drawnCards.add(drawn);
+             Ui.cls();
+             Ui.println("You drew a " + drawn.getName());
+             Ui.pause();
+
+             // Shows current cards and drawn cards
+             Ui.cls();
+             Ui.println(getHeader());
+             Ui.println();
+             Ui.print("Cards:");
+             // Draw every card
+             for(int i = 0; i < drawnCards.size(); i++){
+                 Ui.print(" " + drawnCards.get(i));
+                 if(i+1 < drawnCards.size()){
+                     Ui.print(",");
+                 }
+             }
+             // Value of cards
+             Ui.println("Value: " + highestPossibleValue(drawnCards));
+             Ui.println();
+
+             // Check for too much points
+             if(highestPossibleValue(drawnCards) > 21){
+                 Ui.println("Unfortunately you got more than 21 points, resulting in a total of 0 points!");
+                 Ui.pause();
+                 break;
+             }else {
+                 Ui.println("Would you like to draw another card?");
+                 Ui.println("1) Take another card");
+                 Ui.println("2) Stop taking cards");
+             }
+
+             // Input
+             boolean quitInput = false;
+             while(!quitInput) {
+                 int menuChoice = Ui.getValidInt();
+
+                 switch (menuChoice) {
+                     case 1:
+                         quitInput = true;
+                         break;
+                     case 2:
+                         exit = true;
+                         quitInput = true;
+                         break;
+                     default:
```

Comment:
I saw this structure in some of your code. I thought it would be a good idea to continue this way. Consistency is important, so I'll let it stay this way :)